### PR TITLE
6353 - use authProvider, authRoutes and isApiReady from BCL

### PR DESCRIPTION
**Pull request by alibombaye**
[AB#6699](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6699)

### CODE REVIEW

File: app/helpers/createAuthProvider.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const { AuthProvider, ErrorContext } = require('buying-catalogue-library');
+const { logger } = require('../logger');
+
+export const createAuthProvider = ({ config }) => {
+  const unauthenticatedError = new ErrorContext({
+    status: 401,
+    title: 'You\'re not authorised to view this page',
+    description: 'You must be logged in as a buyer to access Buying Catalogue orders.',
```
Comment (kat1906) [inline app/helpers/createAuthProvider.js:8]: this description need changed as it is admin not OF

File: app/helpers/createAuthProvider.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    backLinkHref: config.publicBrowseBaseUrl,
+  });
+  const authProvider = new AuthProvider({
+    config, scopes: 'Ordering', unauthenticatedError, logger,
```
Comment (kat1906) [inline app/helpers/createAuthProvider.js:12]: and this?

Comment (alibombaye) [reply app/helpers/createAuthProvider.js:12]: ooops very good spot!!!

File: app/helpers/createAuthProvider.js (hunk 1 above)
Comment (alibombaye) [reply app/helpers/createAuthProvider.js:8]: yes and this too.. yikes rushing it

File: app/helpers/createAuthProvider.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const { AuthProvider, ErrorContext } = require('buying-catalogue-library');
+const { logger } = require('../logger');
+
+export const createAuthProvider = ({ config }) => {
+  const unauthenticatedError = new ErrorContext({
+    status: 401,
+    title: 'You\'re not authorised to view this page',
+    description: 'You must be logged in as an admin to access Buying Catalogue admin.',
```
Comment (kat1906) [inline app/helpers/createAuthProvider.js:8]: `You must be logged in as an admin USER to access Buying Catalogue admin.`?

File: app/helpers/createAuthProvider.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const { AuthProvider, ErrorContext } = require('buying-catalogue-library');
+const { logger } = require('../logger');
+
+export const createAuthProvider = ({ config }) => {
+  const unauthenticatedError = new ErrorContext({
+    status: 401,
+    title: 'You\'re not authorised to view this page',
+    description: 'You must be logged in as an admin USER to access Buying Catalogue admin.',
```
Comment (kat1906) [inline app/helpers/createAuthProvider.js:8]: not with capitals 🤣 I wrote it like that to make it obvious what I changed

Comment (alibombaye) [reply app/helpers/createAuthProvider.js:8]: 😂

**Pull request merged**